packages/react/src/Delay.tsx

```diff
@@ -1,12 +1,22 @@
-import { type PropsWithChildren, type ReactNode, useContext, useState } from 'react'
+import { type ReactElement, type ReactNode, useContext, useState } from 'react'
 import { DelayDefaultPropsContext } from './contexts'
 import { useTimeout } from './hooks'
 import { Message_Delay_ms_prop_should_be_greater_than_or_equal_to_0, SuspensiveError } from './models/SuspensiveError'
 
-export interface DelayProps extends PropsWithChildren {
-  ms?: number
-  fallback?: ReactNode
-}
+export type DelayProps =
+  | {
+      ms?: number
+      fallback?: never
+      /**
+       * @experimental This is experimental feature.
+       */
+      children?: ({ isDelayed }: { isDelayed: boolean }) => ReactElement
```

Comment:
I can't say this is the best approach, but it seems it can also be expressed through [ChildrenRenderProps](https://github.com/toss/suspensive/blob/main/packages/jotai/src/utility-types/ChildrenRenderProps.ts).

Reply:
I'll make return as ReactNode! thanks!